### ISSUE TITLE
additional filters changes

### ISSUE THREAD

**Issue by jamerlan** [springlobby/springlobby#328]
1) make "Status" checkboxes work using "OR" not "AND". For example, if I check 2 of them - nothing will be displayed. Its a bug! (tested with latest git version 2daefe702bc71ba7141f74fd7083d56cdd6b6248) on 1 feb 2015

2) By default, filters should show all games. So when player activates it - it shows all the battles by default. Then player can check some checkboxes or select smth in comboboxes and only then results should be narrowed. Checkboxes like "only maps I have" should be unchecked.
![more_filters_changes](https://cloud.githubusercontent.com/assets/216801/5992927/f5920570-aa4e-11e4-884f-0ab6cbe1f0a3.png)

**Comment by Usaga** [comment]
All checkboxes except "open" works in "AND" logic and seems doing it well (checked Locked and Full shows only Locked AND Full games)
"Open" checkbox shows games that are (NOT locked AND NOT passworded and so on...) AND (Only maps.. AND Only mods..). This was kind of weird to place such checkbox with others, but no one complained before))
I can make all checkoboxes inactive and not involved in filtering when "open"  is checked. Or I misunderstood your proposal?

**Comment by jamerlan** [comment]
My example: I want to see all games: Open, Full and Started so I check all three checkboxes. Result - battle list is empty. That is why I said "it uses AND logic". 
more details in russian: Похоже, что внутри ряда чекбоксов, используется AND логика. А я бы хотел, чтобы было OR. Мне кажется, что интуитивно понятнее, когда отметил галочку - значит показывать то, что попадает под этот критерий. Если бы можно было выбрать только одно, то была бы радиокнопка, а не чекбокс. 
А насчет чекбоксов "Only maps I have" - мне кажется, что по умолчанию, они должны быть выключены, чтобы новичкам было проще. Когда активируешь фильтр, и сразу же все исчезает, хотя в фильтре ты еще не успел ничего поменять - это выглядит странно. 
У меня вообще интересная ситуация получилась. У меня был включен фильтр. После обновления, мой фильтр все еще остался активным, но или логика поменялась или чекбоксы стали включены... вобщем, у меня был пустой список битв. Я сначала не понял, что это из-за фильтров. Перезапускал лобби несколько раз, не мог понять, куда все подевалось.

**Comment by Usaga** [comment]
My poor language skill does not allow me to explain it clear. The filter shows you right exactly what you choose in checkboxes with AND logic. If you check all checkboxes it will be similar like file is disabled (all battles are shown), if you check no checkboxex, you will get nothing in battle list. That is behavior of all checkboxes except "OPEN". Its purpes is to show you battles you can frelly join (batttles not passworded, not started, not fulled up and so on). Logic of this checkbox alters logic of other checkboxes. The battle can not be "OPENED" and "FULL" at the same time. So saying it ease: battle is "opened" if you can join it. If you can not, then battle is not "opened".

The same in russian:
Логика опции "открыто" идёт в разрез с логикой остальных опций. Я не знаю зачем её поставили в один ряд со всеми и даже никак не выделили. Но, в любом случае, эта опция делает всё правильно как и все остальные (теперь уже). По хорошему, эту опцию (открыто) нужно как-то выделить или убрать куда-нибудь в другое место. Она показывает только те игры, к которым ты сможешь присоединиться. Если игра заполнена или закрыта паролем или залочена, то ты не сможешь к ней подключиться, значит игра не "открыта".
Галочки с "карты" и "моды" я поубираю, тут ты всё правильно сказал. А глюки у тебя были с лобби, потому, что я исправил логику двух опций (они что-попало фильтровали). Как-то так)

**Comment by jamerlan** [comment]
I think we still need a solution for this "Open" checkbox. Because its very confusing. For example: if user clicks "Open" then checkboxes "Started" and "Full" should be unchecked automatically.

**Comment by abma** [comment]
if selections are mutual exclusive a radiobox should be used:

http://docs.wxwidgets.org/trunk/appear-radiobox-gtk.png

**Comment by abma** [comment]
ah, i guess thats how it was supposed to work: selecting the checkbox "open" should filter out open games, the same for locked  and full games. i personally never investigated filters because i don't use them. i only use sorting by column in the battlelist.

**Comment by Usaga** [comment]
No. I think now that jamerlan was right and most user-friedly behaviour is to filter out games not matching at least one of selected options, except ""Only maps I have" and "Only games I have"...

**Comment by specing** [comment]
Is this fixed now abma?

**Comment by abma** [comment]
idk, i never used filters :-|
